evm/src/executor/inspector/cheatcodes/ext.rs

```diff
@@ -160,21 +159,46 @@ fn get_env(key: &str, r#type: ParamType, delim: Option<&str>) -> Result<Bytes, B
         .map_err(|e| e.into())
 }
 
-fn read_file(path: &str) -> Result<Bytes, Bytes> {
-    let data = fs::read_to_string(path).map_err(|err| err.to_string().encode())?;
+macro_rules! check_path_allowed {
+    ($state:ident, $absolute_path:ident) => {
+        // Check if any of the allowed paths is the prefix for our absolute path.
+        if !$state
+            .config
+            .allowed_paths
+            .iter()
+            .any(|allowed_path| $absolute_path.starts_with(allowed_path))
+        {
+            return Err(util::encode_error("Path is not allowed."))
+        }
+    };
+}
+
+fn full_path(state: &mut Cheatcodes, path: impl AsRef<Path>) -> PathBuf {
+    state.config.root.join(path)
+}
+
+fn read_file(state: &mut Cheatcodes, path: impl AsRef<Path>) -> Result<Bytes, Bytes> {
```

Comment:
I'm a little torn regarding this style

usually, I'd prefer making this `Cheatcodes::read_file(&mut self`, but since we're in a submodule here, maybe this is more fitting. 
However, I believe this is consistent with how functions in the other cheatcodes sub modules are called so lgtm.